cloud/shared/bin/python_env_setup

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env bash
+#
+# The purpose of this script is to dynamically enable python packages.
+# The current main use case is the usage of the env-var-docs parser, which is defined
+# in the civiform repository.
+
+# Initializes a python virtual environment(venv) and installs required dependencies.
+# Takes an argument that specifies the path to the requirements.txt file,
+# which contains a list of python packages to be installed in the venv.
+# For more details about requirement files see:
+# https://pip.pypa.io/en/stable/user_guide/#requirements-files
+function initialize_python_env() {
+
+  echo "initializing python env from script"
+
+  local requirements_file_path=$1
+
+  # Check if there are any requirements to install
+  if [[ ! -f "$requirements_file_path" ]]; then
+    echo "Requirements file $requirements_file_path does not exist, python setup skipped"
+    return
+  fi
+
+  # Create a virtual environment if it doesn't exist yet
+  if [[ ! -d .venv ]]; then
```

Comment:
The scripts in civiform-deploy remove the checkout directory on every run: https://github.com/civiform/civiform-deploy/blob/main/bin/lib/checkout.sh#L80.  This was due to the checkout directory getting larger on each run: https://github.com/civiform/civiform-deploy/pull/31.

If we put the venv directory in the checkout directory, with the current civiform-deploy scripts, we will re-download the env-var-docs parser package on every deployment tool run.  This adds 5-10 seconds of latency to every run.  What do you think about modifying the civiform-deploy script to not remove the .venv directory?

Reply:
Good catch, I'll look into it in a separate PR as it needs to be done in the other repository.
We could either exclude the .venv from the removal or place the .venv elsewhere, I'll have a think about it.

Reply:
Ps: as the PR does not yet enable the feature, the latency won't have an impact.